tests/test_rechunk.py

```diff
@@ -131,6 +242,86 @@ def test_rechunk_dataset(
     assert ds.attrs == dst.attrs
 
 
+@pytest.mark.parametrize("shape", [(100, 50)])
+@pytest.mark.parametrize("source_chunks", [(10, 50), (100, 5)])
+@pytest.mark.parametrize(
+    "target_chunks",
+    [
+        {"x": 20},  # ? Should this rechunk y? Probably not...
+        {"x": 20, "y": 1e5},
+        {"x": 20, "y": -1},
+    ],
+)
+@pytest.mark.parametrize("max_mem", ["10MB"])
+@pytest.mark.parametrize("target_store", ["target.zarr", "mapper.target.zarr"])
+@pytest.mark.parametrize("temp_store", ["temp.zarr", "mapper.temp.zarr"])
+def test_rechunk_dataset_dimchunks(
+    tmp_path, shape, source_chunks, target_chunks, max_mem, target_store, temp_store,
+):
+    if target_store.startswith("mapper"):
+        fsspec = pytest.importorskip("fsspec")
+        target_store = fsspec.get_mapper(str(tmp_path) + target_store)
+        temp_store = fsspec.get_mapper(str(tmp_path) + temp_store)
+    else:
+        target_store = str(tmp_path / target_store)
+        temp_store = str(tmp_path / temp_store)
+
+    a = numpy.arange(numpy.prod(shape)).reshape(shape).astype("f4")
+    a[-1] = numpy.nan
```

Comment:
What is this line for?

Reply:
No clue. Literally just copied that test from the existing ones.

Reply:
Rather than copying this somewhat mysterious logic into a new test, let's at least move it into a fixture so we can eventually fix it once and for all.